tests/libtest.sh

```diff
@@ -540,6 +540,29 @@ skip_without_user_xattrs () {
     fi
 }
 
+# https://brokenpi.pe/tools/strace-fault-injection
+_have_strace_fault_injection=''
+have_strace_fault_injection() {
+    if test "${_have_strace_fault_injection}" = ''; then
+        if strace -P ${test_srcdir}/libtest-core.sh -e inject=read:retval=0 cat ${test_srcdir}/libtest-core.sh >out.txt &&
+           test '!' -s out.txt; then
+            _have_strace_fault_injection=0
```

Comment:
I think I would change this to the less confusing `=1` and `=0` in the `else` branch, and then make the last line of the function `test "${_have_strace_fault_injection}" = '1'`.